rqpy/io/_save.py

```diff
@@ -136,5 +139,57 @@ def saveevents_midgz(events, settings, savepath, savename, dumpnum):
     mywriter.open_file(filename_out, savepath)
     mywriter.write_settings_from_dict(settings)
     mywriter.write_events(events)
-    mywriter.close_file()
+    mywriter.close_file()  
+
+
+def convert_midgz_to_h5(path, savepath, channels, det, lgcskip_empty=False):
+    """
+    Function to convert raw traces and event numbers for a single dump from mid.gz to HDF5.
+    
+    Saves the ndarray of traces, converted to units of TES current, and saves a corresponding
+    array of event numbers. Note, since event numbers are not unique, the series number is
+    appended to the front of the event number, ie. seriesnumber_eventnumber (as an integer
+    withough the underscore) 
+    
+    Parameters
+    ----------
+    path : str, list of str
+        Absolute path, or list of paths, to the dump to open. 
+    savepath : str
+        Absolute path to where the dump should be saved
+    channels : str, list of str
+        Channel name(s), i.e. 'PDS1'. If a list of channels, the outputted traces will be sorted to match the order
+        the getRawEvents reports in events[det]['pChan'], which can cause slow downs. It is recommended to match
+        this order if opening many or large files.
+    det : str, list of str
+        Detector name, i.e. 'Z1'. If a list of strings, then should each value should directly correspond to 
+        the channel names. If a string is inputted and there are multiple channels, then it 
+        is assumed that the detector name is the same for each channel.
+    lgcskip_empty : bool, optional
+        Boolean flag on whether or not to skip empty events. Should be set to false if user only wants the traces.
+        If the user also wants to pull extra timing information (primarily for live time calculations), then set
+        to True. Default is False.
+    
+    Returns
+    -------
+    None
+    
+    """
+    
+    if not isinstance(path, list):
+        path = [path]
+
+    savename = []
+    for p in path:
+        savename.append(p.split('/')[-1].split('.')[0])
+
+    for jj, p in enumerate(path):
+        x, info_dict = get_traces_midgz(p, channels, det, lgcskip_empty=False, lgcreturndict=True)
```

Comment:
You don't actually pass the lgcskip_empty and lgcreturndict parameters to get_traces_midgz

Reply:
opps